### ISSUE TITLE
Bug:  Type Undefined to DIY IPTV

### ISSUE THREAD

**Issue by jey3dayo** [OpenWonderLabs/homebridge-switchbot#545]
### Describe The Bug

can't support DIY IPTV. It was working in v1.x.x.

### To Reproduce

push TV button.

### Expected behavior

The TV turns on.

### Relevant log output

```shell
debug log(Turn ON TV)


[10/20/2022, 3:13:54 AM] [@switchbot/homebridge-switchbot] This plugin generated a warning from the characteristic 'Saturation': characteristic value expected valid finite number and received "undefined" (undefined). See https://homebridge.io/w/JtMGR for more info.
[10/20/2022, 3:13:55 AM] [SwitchBot] [DEBUG] DIY IPTV: MyTV Active: 1
[10/20/2022, 3:13:55 AM] [SwitchBot] [DEBUG] DIY IPTV: MyTV pushTvOnChanges Active: 0, allowPushOn: false
[10/20/2022, 3:13:55 AM] [SwitchBot] DIY IPTV: MyTV Connection Type: undefined, commands will not be sent to OpenAPI
```
```


### Config for homebridge-switchbot

```shell
Not set.
```

### Screenshots

device info
![image](https://user-images.githubusercontent.com/16203828/196774616-5f7590d1-b64c-46d1-845d-224e95cda3ba.png)

### Device & Model

switch bot hub mini

### Node.js Version

v16.14.0

### NPM Version

v8.3.1

### Homebridge Version

v4.50.1

### Homebridge Switchbot Plugin Version

v2.2.1

### Operating System

Docker

**Comment by jey3dayo** [comment]
Might be related.
https://github.com/OpenWonderLabs/homebridge-switchbot/issues/507

**Comment by jey3dayo** [comment]
Maybe something is wrong with the IR devices.
Air conditioner doesn't work.

```
[10/20/2022, 3:34:55 AM] [@switchbot/homebridge-switchbot] This plugin generated a warning from the characteristic 'Saturation': characteristic value expected valid finite number and received "undefined" (undefined). See https://homebridge.io/w/JtMGR for more info.
[10/20/2022, 3:34:59 AM] [SwitchBot] [DEBUG] Air Conditioner: Air Conditioner Set Active: 0
[10/20/2022, 3:34:59 AM] [SwitchBot] [DEBUG] Air Conditioner: Air Conditioner pushAirConditionerOffChanges, Active: undefined
[10/20/2022, 3:34:59 AM] [SwitchBot] [DEBUG] Air Conditioner: Air Conditioner pushAirConditionerOffChanges Active: undefined, allowPushOff: undefined
```

**Comment by donavanbecker** [comment]
Can you try the latest beta: 

[![npm version](https://badgen.net/badge/homebridge-switchbot@beta/>=2.2.2-beta.7/orange)](https://www.npmjs.com/package/@switchbot/homebridge-switchbot)

[Click here to find out how to install Beta](https://github.com/OpenWonderLabs/homebridge-switchbot/wiki/Beta-Version)

**Comment by jey3dayo** [comment]
@donavanbecker I tried it. It didn't work in v2.2.2-beta.9.

## debug log
### DIY IPTV
```
[10/20/2022, 1:00:25 PM] [SwitchBot] [DEBUG] DIY IPTV: TV Active: 1
[10/20/2022, 1:00:25 PM] [SwitchBot] [DEBUG] DIY IPTV: TV pushTvOnChanges Active: 0, allowPushOn: false
[10/20/2022, 1:00:25 PM] [SwitchBot] DIY IPTV: TV Connection Type: undefined, commands will not be sent to OpenAPI
```

### TV
```
[10/20/2022, 1:01:26 PM] [SwitchBot] [DEBUG] TV: SonyTV Active: 1
[10/20/2022, 1:01:26 PM] [SwitchBot] [DEBUG] TV: SonyTV pushTvOnChanges Active: 0, allowPushOn: false
[10/20/2022, 1:01:27 PM] [SwitchBot] TV: SonyTV Connection Type: undefined, commands will not be sent to OpenAPI
```


## Device Info

### DIY IPTV
![image](https://user-images.githubusercontent.com/16203828/196853880-bcc3626d-2a80-4f71-9f41-4a0bfc7220a1.png)

### TV
![image](https://user-images.githubusercontent.com/16203828/196853718-63826db0-7a01-4c0e-a054-d0ce57ee9ccd.png)

**Comment by donavanbecker** [comment]
Can you try the latest beta: 

[![npm version](https://badgen.net/badge/homebridge-switchbot@beta/>=2.2.2-beta.13/orange)](https://www.npmjs.com/package/@switchbot/homebridge-switchbot)

[Click here to find out how to install Beta](https://github.com/OpenWonderLabs/homebridge-switchbot/wiki/Beta-Version)

**Comment by jey3dayo** [comment]
@donavanbecker It didn't work in v2.2.2-beta.19.

```
[10/21/2022, 4:50:56 PM] [@switchbot/homebridge-switchbot] This plugin generated a warning from the characteristic 'Saturation': characteristic value expected valid finite number and received "undefined" (undefined). See https://homebridge.io/w/JtMGR for more info.
[10/21/2022, 4:50:56 PM] [@switchbot/homebridge-switchbot] Error: 
    at Saturation.Characteristic.characteristicWarning (/usr/local/lib/node_modules/homebridge/node_modules/hap-nodejs/src/lib/Characteristic.ts:2142:105)
    at Saturation.Characteristic.validateUserInput (/usr/local/lib/node_modules/homebridge/node_modules/hap-nodejs/src/lib/Characteristic.ts:2030:14)
    at Saturation.<anonymous> (/usr/local/lib/node_modules/homebridge/node_modules/hap-nodejs/src/lib/Characteristic.ts:1534:24)
    at step (/usr/local/lib/node_modules/homebridge/node_modules/tslib/tslib.js:144:27)
    at Object.next (/usr/local/lib/node_modules/homebridge/node_modules/tslib/tslib.js:125:57)
    at fulfilled (/usr/local/lib/node_modules/homebridge/node_modules/tslib/tslib.js:115:62)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
```

**Comment by donavanbecker** [comment]
Fixed in v2.2.2
